scheduler/src/index.ts

```diff
@@ -33,49 +33,38 @@ const server = app.listen(port, async () => {
     res.header('Content-Type', 'application/json')
     res.json(Scheduling.getAllJobs())
   })
-})
 
-let datasourcePollingJob: schedule.Job
-let pipelinePollingJob: schedule.Job
+  server = app.listen(port, () => {
+    console.log(`Listening on port ${port}`)
+  })
+}
+
+main()
+  .catch(error => {
+    console.error(`Failed to start notification service: ${error}`)
+  })
 
-async function updateDatsources (): Promise<void> {
-  try {
-    return Scheduling.updateDatasources()
-  } catch (e) {
-    return console.log(e)
-  }
+function updateDatasources (): void {
+  Scheduling.updateDatasources()
+    .catch(error => {
+      console.log(error)
+    })
 }
 
 async function initJobs (retries = 30, retryBackoff = 3000): Promise<void> {
-  await Scheduling.initializeJobs(retries, retryBackoff)
-    .catch(() => {
-      console.error('Scheduler: Initialization failed. Shutting down server...')
-      server.close()
-      process.exit()
-    })
+  await Scheduling.initializeJobsWithRetry(retries, retryBackoff)
 
   datasourcePollingJob = schedule.scheduleJob(
     'DatasourceEventPollingJob',
     CHRONJOB_EVERY_2_SECONDS,
-    updateDatsources)
+    updateDatasources)
 }
 
-// log all promise rejections that happen
-// (mostly because they happen in async and don't log the point where it happened)
-// For more information see https://nodejs.org/api/process.html#process_event_unhandledrejection
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-process.on('unhandledRejection', (reason: Error | any, promise: Promise<any>) => {
```

Comment:
Why remove it?
I would agree that it promotes not handling errors correctly which is not something we should encourage...

Reply:
It should not be needed anymore with my changes. This was needed because in some places a function with a return type of `Promise<void>` was passed into a callback of type `(...) => void` which obviously does not handle the returned Promise:
https://github.com/jvalue/open-data-service/blob/5c24de8e6741d6bc5259f01c21df88571fc1110a/scheduler/src/scheduling.ts#L142-L144

The new linter rules will not allow this and will force everyone to handle the Promise appropiate. This means the rejected Promise case must be handled and the easiest way is to just add a `.catch(error => ...)` statement:
```ts
  const scheduledJob = schedule.scheduleJob(`Datasource ${datasourceId}`, executionDate, () => {
    execute(datasourceConfig)
      .catch(error => console.log('Failed to execute job:', error))
  })
```

Reply:
Nice, thanks for the explanation!